dev/cmd/dev.py

```diff
@@ -1,40 +1,39 @@
-import argparse
-import os
-import sys
-
+import glob
+import os.path
 import spack
 import spack.cmd
-import llnl.util.tty as tty
 
-SPACKDEV_FILE = os.path.realpath(os.path.expanduser(__file__))
-SPACKDEV_LIB = os.path.realpath(os.path.join(os.path.dirname(SPACKDEV_FILE),
-                                             '..', '..', 'lib'))
-sys.path.insert(0, SPACKDEV_LIB)
+# Import sub-commands
+subcommands = []
+
+
+def import_subcommands_automatically():
+    subcommand_glob = os.path.join(os.path.dirname(__file__), '..', 'subcommands', '*.py')
+    command_files = glob.glob(subcommand_glob)
+    subcommands.extend(filter(
+        lambda x: not x.startswith('__'), [os.path.basename(x)[:-3] for x in command_files]
+    ))
+    t = __import__('subcommands', globals(), locals(), subcommands, 2)
+    for name in subcommands:
+        globals()[name] = getattr(t, name)
+
+import_subcommands_automatically()
 
 description = 'develop multiple Spack packages simultaneously'
 section = 'developer'
 level = 'long'
 
-_subcmd_dir = os.path.join(SPACKDEV_LIB, 'fnal', 'spack', 'dev', 'cmd')
-_subcmds = None
 _subcmd_functions = {}
 
 
-def add_subcommand(subparser, subcmd):
-    pname = spack.cmd.python_name(subcmd)
-    module = spack.cmd.get_module_from(pname, 'fnal.spack.dev')
-    sp = subparser.add_parser(subcmd, help=module.description)
-    module.setup_parser(sp)
```

Comment:
@chissg I'm re-reading the description of spack/spack/11145:

> Enable the encapsulation of sub-commands to top level Spack or extension commands, removing the necessity to explicitly add sub-command parser and implementation invocations to the top level command.

Please note that this is true also for the case on the right, showing what can be done right now without modifications to Spack - so I think **enable** would not be the correct term. Can you please update the description or comment if you disagree?